core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -206,6 +210,11 @@ private[spark] object Task {
       dataOut.writeLong(timestamp)
     }
 
+    // Write the task properties separately so it is available before full task deserialization.
+    val propBytes = Utils.serialize(task.localProperties)
```

Comment:
Hmm, good point. `Utils.serialize` is fine here, since it doesn't matter whether we use a custom serializer here and because today it's always going to be `JavaSerializer` anyways.